src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,324 @@
+    # We create this folder and call it `fastai_path`
+    fastai_path = os.path.join(learner.path, save_directory)
+    os.makedirs(fastai_path, exist_ok=True)
+    learner.export(
+        fname=os.path.join(save_directory, "model.pkl"), pickle_protocol=pickle_protocol
+    )
+
+    # We move the model from `self.path/save_directory/model.pkl` to `save_directory/model.pkl`.
+    os.rename(
+        os.path.join(fastai_path, "model.pkl"),
+        os.path.join(save_directory, "model.pkl"),
+    )
+
+
+def from_pretrained_fastai(
+    model_id: str,
+    config: Optional[Dict] = None,
+    revision: Optional[str] = None,
+    cache_dir: Optional[Union[str, Path]] = None,
+):
+    """
+    Load `model_id` files from the Hub.
+
+    Parameters:
+        model_id (:obj:`str`):
+            The model id where the pickled fastai.Learner is. Example: 'espejelomar/fastai-pet-breeds-classification'.
+        config (:obj:`dict`, `optional`):
+            Configuration object.
+        revision (:obj:`str`, `optional`):
+            Revision at which the repo's files are downloaded. See documentation of `snapshot_download`.
+        cache_dir (:obj:`str`, `Path`, `optional`):
+            Path to cache directory.
+
+    Returns:
+        The `fastai.Learner` model in the `model_id` repo.
+    """
+    # Check that fastai and fastcore versions are supported.
+    check_fastai_fastcore_versions()
+
+    # Import `load_learner` from `fastai.learner`.
+    from fastai.learner import load_learner
+
+    # Root is either a local filepath matching model_id or a cached snapshot
+    if not os.path.isdir(model_id):
+        storage_folder = snapshot_download(
+            repo_id=model_id, revision=revision, cache_dir=cache_dir
+        )
+    else:
+        storage_folder = model_id
+
+    # Loading `model.pkl`.
+    logger.info(
+        f"Using `fastai.Learner` stored in {os.path.join(model_id, 'model.pkl')}."
+    )
+    model = load_learner(os.path.join(storage_folder, "model.pkl"))
+    model.config = config
+    return model
+
+
+def push_to_hub_fastai(
+    learner,
+    repo_id: str,
+    commit_message: Optional[str] = "Add model",
+    private: Optional[bool] = None,
+    use_auth_token: Optional[Union[bool, str]] = True,
+    config: Optional[dict] = None,
+    **kwargs,
+):
+    """
+    Upload learner checkpoint files to the Hub while synchronizing a local clone of the repo in
+    :obj:`repo_id`.
+
+    Parameters:
+        learner (:obj:`Learner`):
+            The `fastai.Learner' you'd like to push to the Hub.
+        repo_id (:obj:`str`):
+            Can either be a repository name for your model in the Hub or a path to a local folder (in
+            which case the repository will have the name of that local folder). The user can be either your individual account (e.g. 'espejelomar/sentece-embeddings-BETO') or an organization you have write access to (e.g. 'stanfordnlp/stanza-de').
```

Comment:
One idea for simplifying the interface is that `repo_id` will just be a repo name of the model in the Hub. I don't think supporting local folder is an important use case, and it's more intuitive.